### PR TITLE
Support file deletion for data-url

### LINKED ISSUE #3957: Files cannot be removed
### Prerequisites

- [X] I have searched the existing issues
- [X] I understand that providing a [SSCCE](http://sscce.org/) example is tremendously useful to the maintainers.
- [X] I have read the [documentation](https://rjsf-team.github.io/react-jsonschema-form/docs)
- [x] Ideally, preferably a [shared playground link](https://rjsf-team.github.io/react-jsonschema-form/#eyJmb3JtRGF0YSI6eyJmaWxlcyI6WyJkYXRhOnRleHQvcGxhaW47bmFtZT1mb28udHh0O2Jhc2U2NCxTR1ZzYkc4aENnPT0iLCJkYXRhOnRleHQvcGxhaW47bmFtZT1mb28udHh0O2Jhc2U2NCxTR1ZzYkc4aENnPT0iXSwiZmlsZSI6ImRhdGE6dGV4dC9wbGFpbjtuYW1lPWZvby50eHQ7YmFzZTY0LFNHVnNiRzhoQ2c9PSJ9LCJzY2hlbWEiOnsidGl0bGUiOiJGaWxlcyIsInR5cGUiOiJvYmplY3QiLCJwcm9wZXJ0aWVzIjp7ImZpbGUiOnsidHlwZSI6InN0cmluZyIsImZvcm1hdCI6ImRhdGEtdXJsIiwidGl0bGUiOiJTaW5nbGUgZmlsZSJ9LCJmaWxlcyI6eyJ0eXBlIjoiYXJyYXkiLCJ0aXRsZSI6Ik11bHRpcGxlIGZpbGVzIiwiaXRlbXMiOnsidHlwZSI6InN0cmluZyIsImZvcm1hdCI6ImRhdGEtdXJsIn19fX0sInVpU2NoZW1hIjp7ImZpbGVzQWNjZXB0Ijp7InVpOm9wdGlvbnMiOnsiYWNjZXB0IjoiLnBkZiJ9fX0sInRoZW1lIjoiZGVmYXVsdCIsImxpdmVTZXR0aW5ncyI6eyJzaG93RXJyb3JMaXN0IjoidG9wIiwiZXhwZXJpbWVudGFsX2RlZmF1bHRGb3JtU3RhdGVCZWhhdmlvciI6eyJhcnJheU1pbkl0ZW1zIjp7InBvcHVsYXRlIjoicG9wdWxhdGUiLCJtZXJnZUV4dHJhRGVmYXVsdHMiOmZhbHNlfSwiZW1wdHlPYmplY3RGaWVsZHMiOiJwb3B1bGF0ZUFsbERlZmF1bHRzIn19fQ==) demonstrating the issue.

### What theme are you using?

core (Actually, I tried them all on the playground.)

### Version

5.13.2

### Current Behavior

There is no way to delete a file, once submitted.  This is true both for single instances and for multiple files (arrays with file items)

### Expected Behavior

- Single files should be deletable
- Files in lists should be deletable.

### Steps To Reproduce

1. Open this [playground example](https://rjsf-team.github.io/react-jsonschema-form/#eyJmb3JtRGF0YSI6eyJmaWxlcyI6WyJkYXRhOnRleHQvcGxhaW47bmFtZT1mb28udHh0O2Jhc2U2NCxTR1ZzYkc4aENnPT0iLCJkYXRhOnRleHQvcGxhaW47bmFtZT1mb28udHh0O2Jhc2U2NCxTR1ZzYkc4aENnPT0iXSwiZmlsZSI6ImRhdGE6dGV4dC9wbGFpbjtuYW1lPWZvby50eHQ7YmFzZTY0LFNHVnNiRzhoQ2c9PSJ9LCJzY2hlbWEiOnsidGl0bGUiOiJGaWxlcyIsInR5cGUiOiJvYmplY3QiLCJwcm9wZXJ0aWVzIjp7ImZpbGUiOnsidHlwZSI6InN0cmluZyIsImZvcm1hdCI6ImRhdGEtdXJsIiwidGl0bGUiOiJTaW5nbGUgZmlsZSJ9LCJmaWxlcyI6eyJ0eXBlIjoiYXJyYXkiLCJ0aXRsZSI6Ik11bHRpcGxlIGZpbGVzIiwiaXRlbXMiOnsidHlwZSI6InN0cmluZyIsImZvcm1hdCI6ImRhdGEtdXJsIn19fX0sInVpU2NoZW1hIjp7ImZpbGVzQWNjZXB0Ijp7InVpOm9wdGlvbnMiOnsiYWNjZXB0IjoiLnBkZiJ9fX0sInRoZW1lIjoiZGVmYXVsdCIsImxpdmVTZXR0aW5ncyI6eyJzaG93RXJyb3JMaXN0IjoidG9wIiwiZXhwZXJpbWVudGFsX2RlZmF1bHRGb3JtU3RhdGVCZWhhdmlvciI6eyJhcnJheU1pbkl0ZW1zIjp7InBvcHVsYXRlIjoicG9wdWxhdGUiLCJtZXJnZUV4dHJhRGVmYXVsdHMiOmZhbHNlfSwiZW1wdHlPYmplY3RGaWVsZHMiOiJwb3B1bGF0ZUFsbERlZmF1bHRzIn19fQ==)
2. Try to remove any of the submitted files as a user.

### Environment

Playground.

### Anything else?

A workaround is to wrap the files into objects: [Playground](https://rjsf-team.github.io/react-jsonschema-form/#eyJmb3JtRGF0YSI6eyJmaWxlcyI6WyJkYXRhOnRleHQvcGxhaW47bmFtZT1mb28udHh0O2Jhc2U2NCxTR1ZzYkc4aENnPT0iLCJkYXRhOnRleHQvcGxhaW47bmFtZT1mb28udHh0O2Jhc2U2NCxTR1ZzYkc4aENnPT0iXSwic3RyaW5ncyI6WyJoZWxsbyIsIndvcmxkIl0sImZpbGVzX3dvcmthcm91bmQiOlt7fV0sImZpbGUiOiJkYXRhOnRleHQvcGxhaW47bmFtZT1mb28udHh0O2Jhc2U2NCxTR1ZzYkc4aENnPT0ifSwic2NoZW1hIjp7InRpdGxlIjoiRmlsZXMiLCJ0eXBlIjoib2JqZWN0IiwicHJvcGVydGllcyI6eyJmaWxlIjp7InR5cGUiOiJzdHJpbmciLCJmb3JtYXQiOiJkYXRhLXVybCIsInRpdGxlIjoiU2luZ2xlIGZpbGUifSwiZmlsZXMiOnsidHlwZSI6ImFycmF5IiwidGl0bGUiOiJNdWx0aXBsZSBmaWxlcyIsIml0ZW1zIjp7InR5cGUiOiJzdHJpbmciLCJmb3JtYXQiOiJkYXRhLXVybCJ9fSwic3RyaW5ncyI6eyJ0eXBlIjoiYXJyYXkiLCJ0aXRsZSI6Ik11bHRpcGxlIHN0cmluZ3MiLCJpdGVtcyI6eyJ0eXBlIjoic3RyaW5nIn19LCJmaWxlc193b3JrYXJvdW5kIjp7InR5cGUiOiJhcnJheSIsInRpdGxlIjoiTXVsdGlwbGUgZmlsZXMgKHdvcmthcm91bmQpIiwiaXRlbXMiOnsidHlwZSI6Im9iamVjdCIsInRpdGxlIjoiTmV4dCBmaWxlOiIsInByb3BlcnRpZXMiOnsiZmlsZSI6eyJ0aXRsZSI6IkVudGVyIHlvdXIgZmlsZSIsInR5cGUiOiJzdHJpbmciLCJmb3JtYXQiOiJkYXRhLXVybCJ9fX19fX0sInVpU2NoZW1hIjp7ImZpbGVzQWNjZXB0Ijp7InVpOm9wdGlvbnMiOnsiYWNjZXB0IjoiLnBkZiJ9fX0sInRoZW1lIjoiZGVmYXVsdCIsImxpdmVTZXR0aW5ncyI6eyJzaG93RXJyb3JMaXN0IjoidG9wIiwiZXhwZXJpbWVudGFsX2RlZmF1bHRGb3JtU3RhdGVCZWhhdmlvciI6eyJhcnJheU1pbkl0ZW1zIjp7InBvcHVsYXRlIjoicG9wdWxhdGUiLCJtZXJnZUV4dHJhRGVmYXVsdHMiOmZhbHNlfSwiZW1wdHlPYmplY3RGaWVsZHMiOiJwb3B1bGF0ZUFsbERlZmF1bHRzIn19fQ==)

### PR DESCRIPTION
**Pull request by yuki-js**
### Reasons for making this change

Fix #3957 
Fix #4033

Before this PR, uploaded files cannot be deleted. 
This PR adds file deletion button, also changes the data flow into the `value` centric one.

Also corrected test descriptions for file widget attributes. The descriptions for two tests in StringField.test.jsx were reversed.

### Checklist

- [ ] **I'm updating documentation**
  - [ ] I've [checked the rendering](https://rjsf-team.github.io/react-jsonschema-form/docs/contributing) of the Markdown text I've added
- [x] **I'm adding or updating code**
  - [x] I've added and/or updated tests. I've run `npm run test:update` to update snapshots, if needed.
  - [ ] I've updated [docs](https://rjsf-team.github.io/react-jsonschema-form/docs) if needed
  - [x] I've updated the [changelog](https://github.com/rjsf-team/react-jsonschema-form/blob/main/CHANGELOG.md) with a description of the PR
- [ ] **I'm adding a new feature**
  - [ ] I've updated the playground with an example use of the feature

### CODE REVIEW

**Comment by heath-freenome** [comment]
@yuki-js Can you explain how Jest was not working? I don't feel comfortable not having tests for this new feature

**Comment by yuki-js** [comment]
Tests became successful (idk why the previous tests failed). I added test cases. Please have a look.

File: CHANGELOG.md [hunk 1]
```diff
@@ -15,6 +15,7 @@ it according to semantic versioning. For example, if your PR adds a breaking cha
 should change the heading of the (upcoming) version to include a major version bump.
 
 -->
+
 # 5.15.3
```
Comment (heath-freenome) [inline CHANGELOG.md:19]: I just switched this to `5.16.0` in my recently merged PR, can you resolve conflicts. Also, it seems much of this file was reformatted. Please revert all of the changes below `5.15.2`. Thanks!

**Comment by heath-freenome** [comment]
@yuki-js can you resolve conflicts and revert most of the unnecessary formatting changes to `CHANGELOG.md`? Thanks!

**Comment by yuki-js** [comment]
I apologize for the oversight. Fixed a changelog, also merged #4033 into this PR.

**Comment by heath-freenome** [comment]
@yuki-js For some reason the build is now failing. Can you please investigate? Thanks

**Comment by yuki-js** [comment]
It's odd that an error happens in v16, and a seemingly unrelated module has failed. Could you rerun the failed job? It might have a timing issue due to Nx's concurrency. Just a heads up.

**Comment by heath-freenome** [comment]
@yuki-js I reran it already. I'll try one more time. In the meantime can you run the build locally and see if it works?

**Comment by heath-freenome** [comment]
Ok, now it works... hmmm

**Review by heath-freenome — APPROVED** [review]

**Review by nickgros — APPROVED** [review]
LGTM

**Pull request merged**